### ISSUE TITLE
wasabi dependency requirement is too relaxed for Ruby 3

### ISSUE THREAD

**Issue by sarahsehr** [savonrb/savon#996]
# Bug report

**Current behavior:**

```
ruby/3.2.1/lib/ruby/gems/3.2.0/gems/wasabi-3.5.0/lib/wasabi/parser.rb:89:in `parse_url': undefined method `unescape' for URI:Module (NoMethodError)

      unescaped_url = URI.unescape(url.to_s)
                         ^^^^^^^^^
```

**Steps to reproduce current behavior:**

1) depend on savon
2) depend on version of wasabi v3.5.0
3) Run app console
4) Call `Savon.client.call`, for example: 

```
Savon.client(wsdl: 'http://www.dneonline.com/calculator.asmx?WSDL').call(:add)
```

**Expected behavior:**

Savon requires that the minimum wasabi version is v3.7.

The cause of this Ruby 3 related error is [removed in v3.6](https://github.com/savonrb/wasabi/blob/master/CHANGELOG.md#360-2020-08-27). However, wasabi was not [tested against any version of Ruby 3 until v3.70](https://github.com/savonrb/wasabi/blob/master/.github/workflows/test.yml).

**System information:**

* ruby version: 3.2.1
* savon version: tested against 2.11.2 and 2.13.0. Note savon v2.14.0 is okay because `savon`'s `httpi` version requirement is incompatible with wasabi v3.5's `httpi` version requirement.

**Additional Context**

I'm filing this bug because it took me time to debug this when upgrading to Ruby 3. Hoping to save someone else the trouble.

**Comment by olleolleolle** [comment]
Would you want to make a PR updating that requirement (which now says 3.4 or higher) to something which avoids defects?
https://github.com/savonrb/savon/blob/master/savon.gemspec#L21C3-L21C40

**Comment by sarahsehr** [comment]
@olleolleolle Sure. Do I need to go through the forking process in the Contributing Guidelines?

**Comment by olleolleolle** [comment]
Closed by #997.
